app/src/main/java/samples/despotoski/nikola/com/bottomsheetsample/BottomSheetActivity.kt

```diff
@@ -0,0 +1,124 @@
+        super.onCreate(savedInstanceState)
+
+        binding = DataBindingUtil.setContentView(this,R.layout.activity_bottom_sheet)
+
+        setSupportActionBar(binding.toolbar)
+
+        setupBottomSheet1()
+
+        setupBottomSheet2()
+    }
+
+    private fun setupBottomSheet2() {
+        binding.showBottomSheet2.setOnClickListener {
+            val bottomSheetDialogFragment: BottomSheetDialogFragment = CustomBottomSheetDialogFragment()
+            bottomSheetDialogFragment.show(supportFragmentManager, bottomSheetDialogFragment.tag)
+        }
+    }
+
+    private fun setupBottomSheet1() {
+        binding.showBottomSheet1.setOnClickListener { //Let's peek it, programmatically
+            mBottomSheetBehavior?.peekHeight = binding.layoutBottomSheet1.textViewDragMe.height
+            binding.layoutBottomSheet1.textViewDragMe.requestLayout()
+        }
+
+        val recyclerView = binding.layoutBottomSheet1.recyclerview
+        recyclerView.layoutManager = LinearLayoutManager(this).apply {
+            isAutoMeasureEnabled = true
+        }
+        recyclerView.adapter = ApplicationAdapter(this, listApplications(this))
+        val parentThatHasBottomSheetBehavior = recyclerView.parent.parent as FrameLayout
+        mBottomSheetBehavior = BottomSheetBehavior.from(parentThatHasBottomSheetBehavior)
+        mBottomSheetBehavior?.let {
+            binding.stateText.setText(getStateAsString(it.state))
+            it.setBottomSheetCallback(object : BottomSheetCallback() {
+                override fun onStateChanged(bottomSheet: View, newState: Int) {
+                    binding.stateText.setText(getStateAsString(newState))
+                }
+
+                override fun onSlide(bottomSheet: View, slideOffset: Float) {
+                    binding.offsetText.text = getString(R.string.offset, slideOffset)
+                }
+            })
+        }
+    }
+
+    override fun onBackPressed() {
+        mBottomSheetBehavior?.let {
+            if (it.state != BottomSheetBehavior.STATE_HIDDEN) {
+                it.setState(BottomSheetBehavior.STATE_COLLAPSED)
+            } else {
+                super.onBackPressed()
+            }
+        }
+
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        menuInflater.inflate(R.menu.menu_bottom_sheet, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        val id = item.itemId
+        return if (id == R.id.action_settings) {
+            true
+        } else super.onOptionsItemSelected(item)
+    }
+
+    companion object {
+        fun getStateAsString(newState: Int): Int {
+            when (newState) {
```

Comment:
This could be come expression body with exhaustive `else -> R.string.undefined` branch